frontend/src/components/shareContent/SocialMediaShareButton.js

```diff
@@ -1,34 +1,64 @@
 import { IconButton, makeStyles } from "@material-ui/core";
 import ShareIcon from "@material-ui/icons/Share";
 import React from "react";
+import { apiRequest } from "../../../public/lib/apiOperations";
 import SocialMediaShareDialog from "./SocialMediaShareDialog";
 
 const useStyles = makeStyles((theme) => ({
-  button: {
-    color: "white",
+  button: (props) => ({
+    color: props.switchColors ? theme.palette.primary.main : "white",
     width: 35,
     height: 35,
-    backgroundColor: theme.palette.primary.main,
+    backgroundColor: props.switchColors ? "white" : theme.palette.primary.main,
     "&:hover": {
-      backgroundColor: theme.palette.primary.main,
+      backgroundColor: props.switchColors ? "white" : theme.palette.primary.main,
     },
-  },
+  }),
 }));
 
 export default function SocialMediaShareButton({
   containerClassName,
-  toggleShowSocials,
-  showSocials,
-  texts,
-  project,
+  contentLinkPath,
+  apiEndpoint,
   locale,
-  projectAdmin,
-  createShareRecord,
-  screenSize,
+  token,
+  messageTitle,
+  tinyScreen,
+  smallScreen,
+  mailBody,
+  texts,
+  dialogTitle,
+  switchColors,
 }) {
-  const classes = useStyles();
+  const classes = useStyles({ switchColors: switchColors });
+
+  const [showSocials, setShowSocials] = React.useState(false);
+  const toggleShowSocials = (value) => {
+    setShowSocials(value);
+  };
 
-  //Assignment of the numbers has to match with SharedProjects.SHARE_OPTIONS in the backend
+  const [linkShared, setLinkShared] = React.useState(false);
+  const createShareRecord = (sharedVia) => {
+    if (sharedVia === 8 && linkShared) return; //only create a share-record for the link once per session
```

Comment:
Is it possible to do this in a different way than hardcoding the id? If not I'd at least create a static variable `SHARED_VIA_LINK_ID` to make clearer what this is about.